### ISSUE TITLE
Model not training smoothly

### ISSUE THREAD

**Issue by EdSu03** [Deepkumar-Pawar/adl-coursework#16]
Accuracy does not increase as expected. Fix bug in architecture/training code.

Following support session:

- [x] Use torchviz to visualise the dependencies on weights
- [x] Adjust hyperparameters
- [x] Use a Sigmoid in the evaluation metric
- [x] Use AUC metric every 2 epochs or so
- [x] Change the AUC evaluation
